addons/website/static/src/components/dialog/seo.js

```diff
@@ -17,8 +19,118 @@ const seoContext = reactive({
     seoName: '',
     metaImage: '',
     defaultTitle: '',
+    updatedAlts: [],
+    brokenLinks: [],
 });
 
+const getSeo = async (self, onlyKeywords = false) => {
+    const pageTextContentEl = self.website.pageDocument.documentElement.querySelector("#wrap");
+
+    const lang = self.state.language ? self.state.language.split("-")[0] : "en";
+    const tagWeights = {
+        h1: 5,
+        h2: 4,
+        h3: 3,
+        a: 2,
+        p: 1,
+    };
+    const maxNGrams = 2;
+
+    const getKeywordsFromText = (text, weight, wordCounts) => {
+        const segmenter = new Intl.Segmenter(lang, { granularity: "word" });
+        const segmentedText = segmenter.segment(text);
+        const words = [...segmentedText]
+            .filter((s) => s.isWordLike)
+            .map((s) => s.segment.toLowerCase())
+            .filter((word) => !/[0-9]+/.test(word));
+        const singleWordsLength = words.length;
+        for (let nGram = maxNGrams; nGram > 1; nGram--) {
+            for (let i = 0; i <= singleWordsLength - nGram; i++) {
+                if (words[i].length > 4 && words[i + nGram - 1].length > 4) {
+                    words.push(words.slice(i, i + nGram).join(" "));
+                }
+            }
+        }
+        if (words) {
+            words
+                .filter((word) => word.length > 4)
+                .forEach((word) => {
+                    if (!wordCounts[word]) {
+                        wordCounts[word] = 0;
+                    }
+                    wordCounts[word] += weight * (word.length - 3);
+                });
+        }
+    };
+
+    const jaccardSimilarity = (str1, str2) => {
+        const set1 = new Set(str1.split(""));
+        const set2 = new Set(str2.split(""));
+
+        const intersection = new Set([...set1].filter((item) => set2.has(item)));
```

Comment:
1. No, I don't need to make sure that the first set is larger. The Jaccard similarity is a symmetric measure, meaning it doesn't depend on the order of the two sets. The calculation of both the intersection and the union is the same, regardless of which set is larger or smaller, so the result will remain consistent.

2. If the intention is to calculate the similarity based on characters alone, then yes, it would make sense to remove spaces from the sets. So I'll filter out these blank spaces in case of bigrams.